packages/core/contracts/financial-templates/contract-for-difference/ContractForDiffrenceCreator.sol

```diff
@@ -0,0 +1,132 @@
+     * @param _tokenFactoryAddress ERC20 token factory used to deploy synthetic token instances.
+     * @param _timerAddress Contract that stores the current time in a testing environment.
+     */
+    constructor(
+        address _finderAddress,
+        address _tokenFactoryAddress,
+        address _timerAddress
+    ) Testable(_timerAddress) nonReentrant() {
+        tokenFactoryAddress = _tokenFactoryAddress;
+        finderAddress = _finderAddress;
+    }
+
+    /**
+     * @param expirationTimestamp unix timestamp of when the contract will expire.
+     * @param collateralPerPair how many units of collateral are required to mint one pair of synthetic tokens.
+     * @param priceIdentifier registered in the DVM for the synthetic.
+     * @param syntheticName Name of the synthetic tokens to be created. The long tokens will have "Long Token" appended
+     *     to the end and the short token will "Short Token" appended to the end to distinguish within the CFD's tokens.
+     * @param syntheticSymbol Symbol of the synthetic tokens to be created. The long tokens will have "l" appended
+     *     to the start and the short token will "s" appended to the start to distinguish within the CFD's tokens.
+     * @param collateralAddress ERC20 token used as as collateral in the CFD.
+     * @param financialProductLibraryAddress Contract providing settlement payout logic.
+     * @notice The created CFD is NOT registered within the registry as the CFD contract uses the DVM.
+     * @notice The CFD constructor does a number of validations on input params. These are not repeated here.
+     */
+    function createContractForDifference(
+        uint64 expirationTimestamp,
+        uint256 collateralPerPair,
+        bytes32 priceIdentifier,
+        string memory syntheticName,
+        string memory syntheticSymbol,
+        address collateralAddress,
+        address financialProductLibraryAddress
+    ) public nonReentrant() returns (address) {
+        // Create a new synthetic token using the params.
+        require(bytes(syntheticName).length != 0, "Missing synthetic name");
+        require(bytes(syntheticSymbol).length != 0, "Missing synthetic symbol");
+        TokenFactory tf = TokenFactory(tokenFactoryAddress);
+
+        // If the collateral token does not have a `decimals()` method, then a default precision of 18 will be
+        // applied to the newly created synthetic token.
+        uint8 collateralDecimals = _getSyntheticDecimals(collateralAddress);
+        ExpandedIERC20 longToken =
+            tf.createToken(
+                string(abi.encodePacked(syntheticName, "Long Token")),
+                string(abi.encodePacked("l", syntheticSymbol)),
+                collateralDecimals
+            );
+        ExpandedIERC20 shortToken =
+            tf.createToken(
+                string(abi.encodePacked(syntheticName, "Short Token")),
+                string(abi.encodePacked("s", syntheticSymbol)),
```

Comment:
Should the short and long tokens share some common symbol to indicate that they are related? This might be too wordy and ugly, but maybe something like `cfdsX` and `cfdlX`. But obviously better name than that haha

Reply:
well, whatever your `syntheticSymbol` and `syntheticName` is will be common between them. is that not sufficient?